### DISCUSSION TITLE
Method for the drivebase.angle() function?

### DISCUSSION THREAD

**Discussion by rotkohl69** [pybricks/support#965] [Q&A] [answered]
Hey guys, I'd like to know how the drivebase estimates the angle it turned. Thanks for your assistance and the effort you guys put into making this awesome project.

**Comment by laurensvalk** [comment, marked as answer]
EDIT: A new answer can be found in https://github.com/pybricks/support/issues/989, so we can close this thread.

Thanks - glad you like this project!

It uses the encoders of the motors to estimate the heading, based on the dimensions you have specified for the wheel diameter and axle track.

 There're no gyro involved (yet).

**Reply by rotkohl69** [reply]
Do you already have an estimate when gyro support is going to be added?

**Reply by laurensvalk** [reply]
Hopefully this year.
